src/config.ts

```diff
@@ -129,16 +129,35 @@ export interface Configuration {
 	stagingDirInput?: string;
 }
 
+const offsetBuildNumber = (
+	buildNumber: string,
+	offset: string | undefined,
+): string => {
+	if (typeof offset === 'undefined') {
+		return buildNumber;
+	}
+	const intOffset = parseInt(offset);
+	const intBuildNumber = parseInt(buildNumber);
+	if (isNaN(intOffset) || isNaN(intBuildNumber)) {
+		return buildNumber;
+	} else {
+		return `${intBuildNumber + intOffset}`;
```

Comment:
Gosh! Could we use the `number` type here? Whilst JS _can_ add two number strings, it shouldn't! 😂

Reply:
I am actually not sure I understand your comment here. I think `intBuildNumber` and `intOffset` are both of type `number` because they are initialised by `parseInt()`, which returns a `number`, is that not right?

We need to convert them from a string at some point because my understanding is that all input args to an action are strings, and also it looks like we support non-numeric build numbers any way as the default value is actually `dev` for the build number (which seems fair enough). That's why I made sure to check for `NaN`.

Reply:
🤦🏽  yep, you're right!

minor: Maybe this'll be more explicit:

```ts
return (intBuildNumber + intOffset).toString()
```

WDYT?

Reply:
Oh yeah that is much better actually!